sphinx/extension.py

```diff
@@ -51,8 +51,9 @@ def verify_needs_extensions(app: Sphinx, config: Config) -> None:
     for extname, reqversion in config.needs_extensions.items():
         extension = app.extensions.get(extname)
         if extension is None:
-            logger.warning(__('The %s extension is required by needs_extensions settings, '
-                              'but it is not loaded.'), extname)
+            msg = __('The %s extension is required by needs_extensions settings, '
+                     'but it is not loaded.')
+            logger.warning(msg, extname)
```

Comment:
@jayaddison would you confirm this is how you proposed formatting it?

and then can you also see, this fails the CI checks:

<img width="1015" alt="image" src="https://github.com/sphinx-doc/sphinx/assets/2997570/5688d519-b720-4537-a179-1924842246f9">

I feel something in your set up is wrong  😅